setup.py

```diff
@@ -19,8 +19,22 @@
 from setuptools import find_packages, setup
 from setuptools.command.test import test as TestCommand
 from version import version
+from pkg_resources import parse_version
 
-toil_version = '3.1.4'
+# Toil version check -- Raise warning instead of using intall_requires to avoid virtualenv conflicts
+toil_min_version = '3.1'
+toil_max_version = '3.2'
```

Comment:
For better or worse, this is the spec that maven (and other JVM build tools, by extension) use
https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html